### PR TITLE
Final part of the API guidelines

### PR DESCRIPTION
**Pull request by mezis**
Last 2, small sections:

- tools of the trade
- further reading

### CODE REVIEW

File: _guidelines/api-design.md [hunk 1]
```diff
@@ -1128,6 +1128,44 @@ maintainability of _our_ software.
...
+part a mounted [Rails engine](http://guides.rubyonrails.org/engines.html) using
+API mode.
+
+Using Sinatra is _not_ recommended as there is no significante performance
+benefit over Rails, it lacks a router (which means all link URLs must be
+manually built), and most non-trivial Sinatra apps end up reinventing MVC.
+
+Using Grape is _not_ recommended as it lacks a router as well.
+
+----------
```
Comment (kapowaz) [inline _guidelines/api-design.md]: Generally speaking HRs are better omitted between headings (they’re kind of unnecessary with the current layout).

Comment (jphastings) [inline _guidelines/api-design.md]: I'll make that change

File: _guidelines/api-design.md [hunk 2]
```diff
@@ -1128,6 +1128,44 @@ maintainability of _our_ software.
...
 
 ## Further reading
 
+- principles:
```
Comment (kapowaz) [inline _guidelines/api-design.md]: I wonder whether these wouldn’t be better as footnotes, that way they can be referenced in-situ with a footnote link that takes you to a descriptive link. At the very least I think these should probably be links.

Comment (jphastings) [inline _guidelines/api-design.md]: :+1: Let me pick that up.

File: _guidelines/api-design.md [hunk 3]
```diff
@@ -1128,6 +1128,44 @@ maintainability of _our_ software.
...
+    - http://en.wikipedia.org/wiki/Representational_state_transfer
+    - http://en.wikipedia.org/wiki/HATEOAS
+
+- example of decently well-thought-out APIs and guidelines
+    - https://github.com/gocardless/http-api-design
+    - https://developer.github.com
+
+- context and debate
+    - http://www.troyhunt.com/2014/02/your-api-versioning-is-wrong-which-is.html
+    - http://devblog.reverb.com/post/47197560134/hal-siren-rabl-roar-garner-building-hypermedia
```
Comment (gbonfant) [inline _guidelines/api-design.md]: I get a 404 on this link

File: _guidelines/api-design.md [hunk 4]
```diff
@@ -1128,6 +1128,44 @@ maintainability of _our_ software.
...
+    - http://restful-api-design.readthedocs.org/
+    - http://en.wikipedia.org/wiki/Representational_state_transfer
+    - http://en.wikipedia.org/wiki/HATEOAS
+
+- example of decently well-thought-out APIs and guidelines
+    - https://github.com/gocardless/http-api-design
+    - https://developer.github.com
+
+- context and debate
+    - http://www.troyhunt.com/2014/02/your-api-versioning-is-wrong-which-is.html
```
Comment (gbonfant) [inline _guidelines/api-design.md]: Nice article but I remember we mention in another guideline we _may_ send version headers in the request, did we end up binning that idea?

File: _guidelines/api-design.md (hunk 2 above)
Comment (kapowaz) [inline _guidelines/api-design.md]: If not footnotes, changing these to subheadings with links like this:

img

…I think works better as a layout.

File: _guidelines/api-design.md [hunk 5]
```diff
@@ -1128,6 +1128,44 @@ maintainability of _our_ software.
...
+    - http://www.foxycart.com/blog/the-hypermedia-debate
+    - http://stackoverflow.com/questions/9055197/splitting-hairs-with-rest-does-a-standard-json-rest-api-violate-hateoas
+
+- more on `application/hal+json`
+    - http://tools.ietf.org/html/draft-kelly-json-hal
+    - http://stateless.co/hal_specification.html
+    - http://alphahydrae.com/2013/06/rest-and-hypermedia-apis/
+
+- spec for json-api
+    - http://jsonapi.org
```
Comment (gbonfant) [inline _guidelines/api-design.md]: A minor comment related to `_links` json-api specifies they can either be a string or an object, we seem to favour the latter, should we add a comment about that?

File: _guidelines/api-design.md [hunk 6]
```diff
@@ -1128,6 +1128,44 @@ maintainability of _our_ software.
...
+
+- context and debate
+    - http://www.troyhunt.com/2014/02/your-api-versioning-is-wrong-which-is.html
+    - http://devblog.reverb.com/post/47197560134/hal-siren-rabl-roar-garner-building-hypermedia
+    - http://www.foxycart.com/blog/the-hypermedia-debate
+    - http://stackoverflow.com/questions/9055197/splitting-hairs-with-rest-does-a-standard-json-rest-api-violate-hateoas
+
+- more on `application/hal+json`
+    - http://tools.ietf.org/html/draft-kelly-json-hal
+    - http://stateless.co/hal_specification.html
```
Comment (gbonfant) [inline _guidelines/api-design.md]: I get a 404 on this as well.

File: _guidelines/api-design.md (hunk 3 above)
Comment (kapowaz) [inline _guidelines/api-design.md]: Looks like the actual URL for this post is https://product.reverb.com/hal-siren-rabl-roar-garner-building-hypermedia-apis-in-ruby-rails-ad73f36fbd84#.xbhz0w44v

File: _guidelines/api-design.md (hunk 6 above)
Comment (kapowaz) [inline _guidelines/api-design.md]: Looks like there’s some sort of top-level issue with that site — visiting the root of the domain you get an error message about the DB.

The article still exists in the [Google cache](http://webcache.googleusercontent.com/search?q=cache:zKvdnlRJp9YJ:stateless.co/hal_specification.html+&cd=1&hl=en&ct=clnk&gl=us&client=safari) though. Maybe it’ll come back?

File: _guidelines/api-design.md (hunk 4 above)
Comment (jphastings) [inline _guidelines/api-design.md]: No, that guideline [still exists](http://deliveroo.engineering/guidelines/api-design/#versioning), this is just for context in the wider debate I believe.

File: _guidelines/api-design.md (hunk 3 above)
Comment (jphastings) [inline _guidelines/api-design.md]: I've removed other 404ing links and changed this one.

File: _guidelines/api-design.md (hunk 6 above)
Comment (jphastings) [inline _guidelines/api-design.md]: I've removed it for now.

File: _guidelines/api-design.md (hunk 5 above)
Comment (jphastings) [inline _guidelines/api-design.md]: Sounds like a good idea :) Want to create a PR for it?

**Pull request merged**